x-pack/plugins/uptime/public/components/monitor/synthetics/step_detail/waterfall/waterfall_filter.tsx

```diff
@@ -0,0 +1,139 @@
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCheckbox,
+  EuiPopover,
+  EuiButtonIcon,
+  EuiSpacer,
+} from '@elastic/eui';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import useDebounce from 'react-use/lib/useDebounce';
+import { FILTER_REQUESTS_LABEL } from '../../waterfall/components/translations';
+import { MimeType } from './types';
+import { OPEN_FILTERS_POPOVER } from '../../translations';
+
+interface Props {
+  query: string;
+  activeFilters: string[];
+  setActiveFilters: Dispatch<SetStateAction<string[]>>;
+  setQuery: (val: string) => void;
+  onlyHighlighted: boolean;
+  setOlyHighlighted: (val: boolean) => void;
+}
+
+export const MIME_FILTERS = [
+  {
+    label: 'XHR',
+    mimeType: MimeType.XHR,
+  },
+  {
+    label: 'HTML',
+    mimeType: MimeType.Html,
+  },
+  {
+    label: 'JS',
+    mimeType: MimeType.Script,
+  },
+  {
+    label: 'CSS',
+    mimeType: MimeType.Stylesheet,
+  },
+  {
+    label: 'Font',
+    mimeType: MimeType.Font,
+  },
+  {
+    label: 'Media',
+    mimeType: MimeType.Media,
+  },
+];
+
+export const WaterfallFilter = ({
+  query,
+  setQuery,
+  activeFilters,
+  setActiveFilters,
+  onlyHighlighted,
+  setOlyHighlighted,
+}: Props) => {
+  const [value, setValue] = useState(query);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const toggleFilters = (val: string) => {
+    setActiveFilters((prevState) =>
+      prevState.includes(val) ? prevState.filter((filter) => filter !== val) : [...prevState, val]
+    );
+  };
+
+  useDebounce(
+    () => {
+      setQuery(value);
+    },
+    250,
+    [value]
+  );
+
+  return (
+    <EuiFlexGroup gutterSize="xs" alignItems="center">
+      <EuiFlexItem>
+        <EuiFieldSearch
+          fullWidth
+          aria-label={FILTER_REQUESTS_LABEL}
+          placeholder={FILTER_REQUESTS_LABEL}
+          onChange={(evt) => {
+            setValue(evt.target.value);
+          }}
+          value={value}
+        />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiPopover
+          button={
+            <EuiButtonIcon
+              aria-label={OPEN_FILTERS_POPOVER}
+              iconType="filter"
+              onClick={() => setIsPopoverOpen((prevState) => !prevState)}
+              color={activeFilters.length > 0 ? 'primary' : 'text'}
+            />
+          }
+          isOpen={isPopoverOpen}
+          closePopover={() => setIsPopoverOpen(false)}
+          anchorPosition="rightCenter"
+        >
+          <EuiFilterGroup>
+            {MIME_FILTERS.map(({ label, mimeType }) => (
+              <EuiFilterButton
+                hasActiveFilters={activeFilters.includes(mimeType)}
+                onClick={() => toggleFilters(mimeType)}
+                key={label}
+                withNext={true}
+              >
+                {label}
+              </EuiFilterButton>
+            ))}
+          </EuiFilterGroup>
+          <EuiSpacer size="xs" />
+          <EuiCheckbox
```

Comment:
I don't think we need this for now.

Reply:
not sure, how chart handle those, nevertheless i have added those where needed.

Reply:
i think for investigative purpose this is really useful to avoid cluttered view.
This will keep domain of the chart while hiding away filtered out items.

Imagine if you want to compare two requests, one might happen on page reload. Or on a click of a button.

Or imagine monitor list view, if it reload, it might sends duplicate requests because of refactoring.

Spotting those in a long scrolled out view would be hard.

In future i think we will add few more knobs like mark/flag duplicated requests etc